File: src/main/java/com/github/sschuberth/stan/data/BookingItem.java
package com.github.sschuberth.stan.data;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class BookingItem {
    public LocalDate postDate;
    public LocalDate valueDate;
    public List<String> info;
    public float amount;

    public BookingItem(LocalDate postDate, LocalDate valueDate, String info, float amount) {
        this.postDate = postDate;
        this.valueDate = valueDate;
        this.info = new ArrayList<>();
        this.info.add(info);
        this.amount = amount;
    }

    @Override
    public String toString() {
        StringBuilder s = new StringBuilder();

        s.append("{\n");

        s.append("\t\"postDate\": \"").append(postDate).append("\",\n");
        s.append("\t\"valueDate\": \"").append(valueDate).append("\",\n");
        s.append("\t\"amount\": \"").append(amount).append("\",\n");

        s.append("\t\"info\": [\n");
        for (String line : info) {
            s.append("\t\t\"");
            s.append(line);
            s.append("\",\n");
        }
        s.append("\t],\n");

        s.append("},\n");

        return s.toString();
    }
}

File: src/main/kotlin/model/BookingItem.kt
package com.github.sschuberth.stan.model

import java.time.LocalDate

class BookingItem(
        val postDate: LocalDate,
        val valueDate: LocalDate,
        infoLine: String,
        val amount: Float
) {
    val info = mutableListOf(infoLine)

    override fun toString(): String {
        val s = StringBuilder("""
            {
            	"postDate": "$postDate",
            	"valueDate": "$valueDate",
            	"amount": "$amount",
            	"info": [

            """.trimIndent()
        )

        val infoIterator = info.iterator()
        while (infoIterator.hasNext()) {
            s.append("\t\t")

            // Quote the info line.
            s.append("\"")
            s.append(infoIterator.next())
            s.append("\"")

            if (infoIterator.hasNext()) {
                s.append(",")
            }

            s.append("\n")
        }

        s.append("\t]\n")
        s.append("}")

        return s.toString()
    }
}
